src/db/mongodb/mod.rs

```diff
@@ -35,12 +35,14 @@ impl MongoDb {
         client_options.app_name = Some("Chronicle".to_string());
         client_options.min_pool_size = config.min_pool_size;
 
-        if let (Some(username), Some(password)) = (&config.username, &config.password) {
-            let credential = Credential::builder()
-                .username(username.clone())
-                .password(password.clone())
-                .build();
-            client_options.credential = Some(credential);
+        if client_options.credential.is_none() {
+            if let (Some(username), Some(password)) = (&config.username, &config.password) {
+                let credential = Credential::builder()
+                    .username(username.clone())
+                    .password(password.clone())
+                    .build();
+                client_options.credential = Some(credential);
+            }
```

Comment:
Should we error or at least log something if both didn't provide any credentials?

Reply:
hmm, probably not necessary as the error will be detailed enough I guess.

Reply:
Yeah, I'd think so.